File: src/models/rate_limit_check_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct RateLimitCheckOut {
    /// Whether the request is allowed
    pub allowed: bool,

    /// Number of tokens remaining
    pub remaining: u64,

    /// Milliseconds until enough tokens are available (only present when allowed is false)
    #[serde(
        rename = "retry_after_ms",
        skip_serializing_if = "Option::is_none",
        with = "crate::duration_ms_serde::optional"
    )]
    pub retry_after: Option<std::time::Duration>,
}

impl RateLimitCheckOut {
    pub fn new(allowed: bool, remaining: u64) -> Self {
        Self {
            allowed,
            remaining,
            retry_after: None,
        }
    }

    pub fn with_retry_after(mut self, value: impl Into<Option<std::time::Duration>>) -> Self {
        self.retry_after = value.into();
        self
    }
}

File: coyote/models/rate_limit_check_out.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class RateLimitCheckOut(BaseModel):
    allowed: bool
    """Whether the request is allowed"""

    remaining: int
    """Number of tokens remaining"""

    retry_after_ms: t.Optional[int] = Field(default=None, alias="retry_after_ms")
    """Milliseconds until enough tokens are available (only present when allowed is false)"""
